tests/Schema/MigratorTest.php

```diff
@@ -400,6 +419,13 @@ public function testIntrospectTableToModelSetPersistence(): void
         self::assertTrue($model->issetPersistence());
         self::assertSame(['id', 'a'], array_keys($model->getFields()));
     }
+
+    public function testJoinFieldsAreNotCreated(): void
+    {
+        $migrator = $this->createMigrator(new TestUserWithJoin($this->db))->create();
+        $tableColumns = $migrator->table->getColumns();
+        self::assertArrayNotHasKey('role_name', $tableColumns);
```

Comment:
```suggestion
        $model = new TestUserWithJoin($this->db);
        $migrator = $this->createMigrator($model);
        self::assertSame(['todo'], array_keys($migrator->table->getColumns()));
        $migrator->create();
        
        $model = $this->createMigrator()->introspectTableToModel('user');
        self::assertSame(['todo'], array_keys($model->getFields()));
```

this will test if the DB fields are constructed correctly and also then if created correctly